invokeai/app/api/routers/model_records.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2023 Lincoln D. Stein
+"""FastAPI route for model configuration records."""
+
+
+from hashlib import sha1
+from random import randbytes
+from typing import List, Optional
+
+from fastapi import Body, Path, Query, Response
+from fastapi.routing import APIRouter
+from pydantic import BaseModel, ConfigDict, TypeAdapter
+from starlette.exceptions import HTTPException
+
+from invokeai.app.services.model_records import DuplicateModelException, InvalidModelException, UnknownModelException
+from invokeai.backend.model_manager.config import AnyModelConfig, BaseModelType, ModelType
+
+from ..dependencies import ApiDependencies
+
+model_records_router = APIRouter(prefix="/v1/model_records", tags=["model_records"])
+
+ModelConfigValidator = TypeAdapter(AnyModelConfig)
+
+
+class ModelsList(BaseModel):
+    """Return list of configs."""
+
+    models: list[AnyModelConfig]
+
+    model_config = ConfigDict(use_enum_values=True)
+
+
+ModelsListValidator = TypeAdapter(ModelsList)
+
+
+@model_records_router.get(
+    "/",
+    operation_id="list_model_configs",
+    responses={200: {"model": ModelsList}},
+)
+async def list_model_records(
+    base_models: Optional[List[BaseModelType]] = Query(default=None, description="Base models to include"),
+    model_type: Optional[ModelType] = Query(default=None, description="The type of model to get"),
+) -> ModelsList:
+    """Get a list of models."""
+    record_store = ApiDependencies.invoker.services.model_records
+    if base_models and len(base_models) > 0:
+        models_raw = list()
+        for base_model in base_models:
+            models_raw.extend(
+                [x.model_dump() for x in record_store.search_by_name(base_model=base_model, model_type=model_type)]
```

Comment:
nit: After reading this, I realized that we could access the DB more efficiently if we update `search_by_name` to accept a list of search values. The 'where' clause would then be something like:

```
...
WHERE base_model in ('model_1', 'model_2) and model_type in ('model_type_1', 'model_type_2') and model_name in ('name_1', 'name_2')
```

I'm fine with leaving this for future optimization though.

Reply:
I like that, but we don't have a use case for it yet, so will leave for a future implementation.